onnxruntime/core/providers/coreml/builders/impl/argmax_op_builder.cc

```diff
@@ -32,57 +32,40 @@ Status ArgMaxOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
 
   NodeAttrHelper helper(node);
   const auto axis = helper.Get("axis", 0);
-  const auto keep_dims = helper.Get("keep_dims", 1);
-  const bool removedim = keep_dims != 1;
+  const auto keepdims = helper.Get("keepdims", 1);
+  const bool removedim = keepdims != 1;
 
   auto* coreml_argmax = layer->mutable_argmax();
   coreml_argmax->set_axis(axis);
   coreml_argmax->set_removedim(removedim);
 
-  // Get ArgMax's next node(Cast)'s outputdefs
-  auto it = node.OutputEdgesBegin();
-  const auto* succ_node(graph_viewer.GetNode(it->GetNode().Index()));
+  // There are two cases here:
+  // 1. Special Case (ArgMax-Cast), we fuse the Argmax's output/Cast's input
```

Comment:
Expected there would unit test updates for the new behavior.

Reply:
AFAIK, the existing single argmax operator UT test should be enough:) For the special case, added new UT test in the last argmax pr. For the newly added more general support, the `ReductionOpTest.ArgMax*` should cover the test.

Reply:
Does the test verify the ArgMax was assigned to CoreML as expected? I understand there are existing unit tests that will check the results, but if something was off and we were unexpectedly falling back to the CPU EP would we notice that?

Reply:
Good point. We now are able to notice that through the log info when running `onnx_test_runner`, we can see which nodes are successfully added to the graph, and which nodes are supported, just as for the other operators.

But not yet a test specifically for ArgMax. Will think about adding one if necessary.